x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/installed_integration_set.ts

```diff
@@ -33,10 +33,59 @@ interface PackageInfo extends InstalledPackageBasicInfo {
 export const createInstalledIntegrationSet = (): IInstalledIntegrationSet => {
   const packageMap: PackageMap = new Map<string, PackageInfo>([]);
 
+  const addPackage = (fleetPackage: PackageListItem): void => {
+    if (fleetPackage.type !== 'integration') {
+      return;
+    }
+    if (fleetPackage.status !== 'installed') {
+      return;
+    }
+
+    const packageKey = `${fleetPackage.name}`;
+    const existingPackageInfo = packageMap.get(packageKey);
+
+    if (existingPackageInfo != null) {
+      return;
+    }
+
+    // Actual `installed_version` is buried in SO, root `version` is latest package version available
+    // SO should be available as PackageListItem should be InstalledRegistry<RegistrySearchResult>
+    const installedPackageVersion = fleetPackage.savedObject?.attributes?.install_version;
+    const latestPackageVersion = fleetPackage.version;
+
+    // Policy templates correspond to package's integrations.
+    const packagePolicyTemplates = fleetPackage.policy_templates ?? [];
+
+    const packageInfo: PackageInfo = {
+      package_name: fleetPackage.name,
+      package_title: fleetPackage.title,
+      package_version: installedPackageVersion ?? latestPackageVersion,
```

Comment:
This fallback is suspicious because `package_version` means the installed version, not the latest. If we want to return the latest version we should do it via a separate property.

Reply:
Yeah, and now with the early return `if (fleetPackage.status !== 'installed')` `installedPackageVersion` should always be present, so I think it's best to remove this as well 👍